CHANGES.md

```diff
@@ -1,6 +1,362 @@
+---------------------------------------------------------
+* updated parser errors to use the new 'dyno' error-reporting framework
+* made `--print-passes` group compilation timings into coarser pass groups  
+  (e.g., 'front-end passes')
+
+Developer-oriented changes: 'dyno' Compiler improvements/changes
+----------------------------------------------------------------
+* improved the directory structure of the 'dyno' source code
+* generally improved integration with the production compiler
+* adjusted the uAST representation for `try` to always include a body `Block`
+* addressed an issue in which parser errors could be issued twice
+* made numerous improvements to 'dyno's scope resolution capabilities:
+  - added support for record and class fields
+  - added support for secondary methods
+  - added support for nested classes and nested procedures
+  - added support for `sync` statements
+  - fixed problems with `require` statements
+  - added proper support for the `private` keyword
+  - added support for modules being discovered twice through distinct `import`s
+  - fixed "undefined variable" errors when a module is imported multiple times
+  - fixed failures when looking up variables inside renamed modules
+  - added support for scope-resolving interfaces
+  - added support for errors indicating when a module is used as a variable
+* updated 'dyno' errors in scope-resolver to use the new reporting framework
+* made `use` statements transitively private by default, as expected
+* fixed `public use` to not bring in the module's name itself
+* added support for detecting conflicting imported symbol names properly
+* improved scope resolution of `except` lists in 'dyno'
+* added initial support for resolving initializers
+* converted several resolution/type errors to use the new error framework
+* added support for enforcing type query constraints in formal arguments
+* added ability to resolve operator overloads
+* added ability to toggle 'dyno' asserts w/ `CHPL_DEVELOPER` or `--[no-]devel`
+* made `--ignore-errors` continue compilation after a 'dyno' assertion failure
+* implemented draft versions of split initialization and copy elision analysis
+
+Developer-oriented changes: Runtime improvements
+------------------------------------------------
+* replaced 'sprintf()' uses with 'snprintf()' in sources
+
+Developer-oriented changes: Platform-specific bug fixes
+-------------------------------------------------------
+* removed a workaround for a problem with Homebrew libc++ based on upstream fix
+* added a check for erroneous OOB information for `CHPL_COMM=ofi`
+
+Developer-oriented changes: Testing System
+------------------------------------------
+* enabled the ability to have a custom environment per test run
+
+Developer-oriented changes: Tool Improvements
+---------------------------------------------
+* factored code from `chpl2rst.py` into `literate_chapel.py`
+
+Developer-oriented changes: Utilities
```

Comment:
Unused section

Reply:
Just making sure you got all the way to the end!  :D